File: scripts/check-repo-baseline.mjs
#!/usr/bin/env node
import { existsSync, readFileSync } from "node:fs";
import { join, resolve } from "node:path";
import { findRepoRoot, parseArgs, readConfig } from "./shared.mjs";

const args = parseArgs();
const root = resolve(args.target || findRepoRoot());
const config = readConfig(root);
const missing = [];

function requirePath(path) {
  if (!existsSync(join(root, path))) missing.push(path);
}

requirePath("README.md");
requirePath("package.json");
requirePath(".unicorn-hub/config.json");
requirePath(config.docsDir || "docs_project");
requirePath(".env.example");
requirePath("docs_project/project/product-brief.md");
requirePath("docs_project/project/architecture.md");

const PROFILE_EXCLUDABLE = new Set([".github/workflows/ci.yml"]);
const requestedExclusions = new Set(config.excludeTemplates || []);
const excluded = new Set([...requestedExclusions].filter((path) => PROFILE_EXCLUDABLE.has(path)));
for (const path of [
  "AGENTS.md",
  "CLAUDE.md",
  ".specify/memory/constitution.md",
  config.specsDir || "specs",
  "scripts/check-context-budget.mjs",
  "scripts/check-feature-memory.mjs",
  "scripts/check-repo-baseline.mjs",
  "scripts/ai-command-policy.mjs",
  "scripts/ai-review-gate.mjs",
  "scripts/ai-review-rerun.mjs",
  "scripts/set-implementation-agent.mjs",
  ".github/workflows/ci.yml",
  ".github/workflows/pr-guard.yml",
  ".github/workflows/ai-review.yml",
  ".github/workflows/ai-review-rerun.yml",
  ".github/workflows/ai-command-policy.yml"
]) {
  if (excluded.has(path)) continue;
  requirePath(path);
}

if (missing.length) {
  console.error("Missing required baseline files:");
  for (const path of missing) console.error(`- ${path}`);
  process.exit(1);
}

const packageJson = JSON.parse(readFileSync(join(root, "package.json"), "utf8"));
if (!packageJson.packageManager?.startsWith("pnpm@")) {
  console.error("package.json must pin packageManager to pnpm@<version>.");
  process.exit(1);
}

console.log("Repository baseline check passed.");

File: scripts/check_repo_baseline.py
from __future__ import annotations

import json
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
REQUIRED_PATHS = [
    "README.md",
    "pyproject.toml",
    ".env.example",
    ".unicorn-hub/config.json",
    "AGENTS.md",
    "CLAUDE.md",
    "docs_project/project/product-brief.md",
    "docs_project/project/architecture.md",
    "specs",
    "src/nome",
    "tests",
    ".github/workflows/ci.yml",
    ".github/workflows/pr-guard.yml",
]
FORBIDDEN_SECRET_PATTERNS = [
    "TELEGRAM_BOT_TOKEN=123",
    ".session",
]


def main() -> int:
    missing = [path for path in REQUIRED_PATHS if not (ROOT / path).exists()]
    if missing:
        print("Missing required repository paths:", file=sys.stderr)
        for path in missing:
            print(f"- {path}", file=sys.stderr)
        return 1

    pyproject = (ROOT / "pyproject.toml").read_text(encoding="utf-8")
    if 'name = "nome"' not in pyproject:
        print("pyproject.toml must define the nome project.", file=sys.stderr)
        return 1

    config = json.loads((ROOT / ".unicorn-hub/config.json").read_text(encoding="utf-8"))
    if config.get("commands", {}).get("preflight") != "uv run python scripts/preflight.py":
        print("Repository config must point preflight at the Python checker.", file=sys.stderr)
        return 1

    tracked_text = "\n".join(
        path.read_text(encoding="utf-8", errors="ignore")
        for path in [
            ROOT / ".env.example",
            ROOT / "README.md",
            ROOT / "AGENTS.md",
            ROOT / "CLAUDE.md",
        ]
    )
    for pattern in FORBIDDEN_SECRET_PATTERNS:
        if pattern in tracked_text:
            print(f"Potential secret/session pattern found: {pattern}", file=sys.stderr)
            return 1

    print("Python repository baseline check passed.")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
